src/editor/editor.js

```diff
@@ -268,6 +270,39 @@ export default class Editor {
 		this.commands.execute( ...args );
 	}
 
+	/**
+	 * Methods creates rejected promise when provided parameter is a HTML Textarea element.
+	 * It resolves in any other case.
+	 *
+	 * @param {HTMLElement|String} sourceElementOrData The DOM element that will be the source for the created editor
+	 * or the editor's initial data.
+	 * @returns {Promise}
```

Comment:
Why this function returns a Promise if it isn't asynchronous itself? It should return `Boolean` or `undefined` and throws an error if specified arguments are invalid.

Reply:
After checking related PR in balloon editor, I am for the second option (`undefined` + throw).

And the entire editor's code could be simplified:

```js
return new Promise( resolve => {	

    Editor.allowedElement( sourceElementOrData ); // Throw the error if specified parameter is invalid.
    // Editor.assertSourceElement() ?
    
    const editor = new this( sourceElementOrData, config );
    // ...
} );
```

Reply:
I use it in 3 editors (balloon, inline, decouple) to test what kind of element is used during editor creation. I felt that simpler is to kept error here rather than implementation of rejected promise in each of those editors (https://github.com/ckeditor/ckeditor5-editor-inline/blob/9b34b1106cc935394a67b4a07bf5a03ca00af5f1/src/inlineeditor.js#L192)
Is it an antipattern to use promise in cases like this?

Reply:
It's more like an assertion check. You would like to check whether the specified parameter is proper from the editor point of view. The similar _mechanism_ we are using in Writer – https://github.com/ckeditor/ckeditor5-engine/blob/e0de1586620918f08854962e3ee323122c4e2059/src/model/writer.js#L1302-L1320.

That's why I am proposing changing the API and usage in `*Editor.create()`  classes.

> Is it an antipattern to use promise in cases like this?

In such case, we are forcing synchronous function to be asynchronous. For me, it sounds bad.